pkg/dependency/parser/python/binary/exe.go

```diff
@@ -0,0 +1,82 @@
+// Ported from https://github.com/golang/go/blob/e9c96835971044aa4ace37c7787de231bbde05d9/src/cmd/go/internal/version/exe.go
```

Comment:
I unfortunately didn't find this in Go 1.22 either. I'll take a look into this in next commit.

Reply:
I have updated the link and also the implementation was changed.

It's not a 1on1 port, since they repurposed it for capturing Go main module version, but we need to read whole executable file, but I used the same ideas. It also contains interfaces for other binaries (didn't put those in, only elf).